src/dialogs/generalsettingsdialog.cpp

```diff
@@ -1,14 +1,53 @@
 #include "generalsettingsdialog.h"
 #include "ui_generalsettingsdialog.h"
+#include <QSettings>
+#include <QMessageBox>
 
 GeneralSettingsDialog::GeneralSettingsDialog(QWidget *parent) :
     QDialog(parent),
     ui(new Ui::GeneralSettingsDialog)
 {
     ui->setupUi(this);
+    connect(ui->buttonBox, &QDialogButtonBox::accepted, this, &GeneralSettingsDialog::saveSettings);
 }
 
 GeneralSettingsDialog::~GeneralSettingsDialog()
 {
     delete ui;
 }
+
+void GeneralSettingsDialog::saveSettings()
+{
+    QSettings settings;
+    QStringList groups;
+
+    settings.beginGroup("GeneralSettings");
+        groups = settings.childGroups();
```

Comment:
The identation here is to facilitate the reading. Don't need to have, however for me it's better.